### PR TITLE
Condition source-build specific configurations on DotNetBuildFromSouce

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to VMR work: https://github.com/dotnet/source-build/issues/3926

Changes are from https://github.com/dotnet/dotnet/pull/46

No functional changes introduced.

### CODE REVIEW

File: eng/Build.props [hunk 1]
```diff
@@ -1,8 +1,7 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
```
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/Build.props:2]: Aah - interesting!

I can still see those in couple of repos in Version.props - so I guess those can go as well, right? (but not an important thing of course)

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply eng/Build.props:2]: Yes those can be removed as well.

**Comment by ViktorHofer** [comment]
I assume the failures in this PR are unrelated?

**Comment by JanKrivanek** [comment]
> I assume the failures in this PR are unrelated?

I do not recall seaing any similar error recently.
Reruning just for sure. Let's have closer look tomorrow if it's persistent

**Comment by JanKrivanek** [comment]
Still failing
`Microsoft.NETCoreSdk.BundledVersions.props` is missing for some reason

image

I'm short on time today - how urgent is this?

**Comment by JanKrivanek** [comment]
Hi @ViktorHofer - I believe the issue is directly related to the change

The missing `Microsoft.NETCoreSdk.BundledVersions.props` is imported in `SourceBuildIntermediate.proj` - that is not part of the failing build:

image

The reason will be likely related to the fact that the conditionaly excluded target (`ConfigureInnerBuildArgs`) prepares `InnerBuildArgs` property that is used later on in other targets. The property is now missing any project to run - so likely rest of the configuration that it's serving is lost that way:

image

I haven't dug deeper into what exactly is the purpose of the property and what is skipped as the result of missing project - but hopefully you are equiped to carry on investigation based on the intent of the change

**Comment by ViktorHofer** [comment]
Thank you for taking a look. This is really weird but let me try something different.

**Review by AR-May — APPROVED** [review]
LGTM

**Review by YuliiaKovalova — APPROVED** [review]

**Pull request merged**